### ISSUE TITLE
Allow praw to automatically refresh tokens

### ISSUE THREAD

**Issue by michael-lazar** [michael-lazar/rtv#139]
PRAW has built-in token refreshing, is there any reason that we can't rely on this instead of implementing our own?

**praw/__init__.py**
```
        while True:
            try:
                self._use_oauth = self.is_oauth_session()
                response = handle_redirect()
                _raise_response_exceptions(response)
                self.http.cookies.update(response.cookies)
                self._use_oauth = tempauth
                if raw_response:
                    return response
                else:
                    return re.sub('&([^;]+);', decode, response.text)
            except errors.OAuthInvalidToken as error:
                if not attempt_oauth_refresh:
                    raise
                attempt_oauth_refresh = False
                self._use_oauth = False
                self.refresh_access_information()
                self._use_oauth = tempauth
                request, key_items, kwargs = build_key_items(url, params,
                                                             data, auth, files,
                                                             method)
```

**Comment by TheoPib** [comment]
I don't understand what you mean here. We already use PRAW's token refreshing, via **refresh_access_information()** method.

**Comment by michael-lazar** [comment]
I mean that we don't need to manually call **refresh_access_information()**. It's already built into every PRAW request. PRAW will check if the token is invalid and refresh automatically.

**Comment by michael-lazar** [comment]
Merged in #141
